artifacts/definitions/Windows/Attack/UnexpectedImagePath.yaml

```diff
@@ -0,0 +1,73 @@
+
+author: Amged Wageh
+
+parameters:
+   - name: expected_paths
+     type: csv
+     default: |
+        ProcName,ExpectedPath
+        csrss.exe,c:\windows\system32\csrss.exe
+        smss.exe,c:\windows\system32\smss.exe
+        services.exe,c:\windows\system32\services.exe
+        wininit.exe,c:\windows\system32\wininit.exe
+        svchost.exe,c:\windows\system32\svchost.exe
+        svchost.exe,c:\windows\syswow64\svchost.exe
+        runtimebroker.exe,c:\windows\system32\runtimebroker.exe
+        lsaiso.exe,c:\windows\system32\lsaiso.exe
+        taskhostw.exe,c:\windows\system32\taskhostw.exe
+        lsass.exe,c:\windows\system32\lsass.exe
+        winlogon.exe,c:\windows\system32\winlogon.exe
+        explorer.exe,c:\windows\explorer.exe
+        explorer.exe,c:\windows\syswow64\explorer.exe
+        conhost.exe,c:\windows\system32\conhost.exe
+        dllhost.exe,c:\windows\system32\dllhost.exe
+        dllhost.exe,c:\windows\syswow64\dllhost.exe
+        wmiprvse.exe,c:\windows\system32\wbem\wmiprvse.exe
+        wmiprvse.exe,c:\windows\syswow64\wbem\wmiprvse.exe
+
+sources:
+  - precondition:
+      SELECT OS From info() where OS = 'windows'
+
+    query: |
+      LET expected_paths_tmp <= SELECT ProcName AS ProcessName FROM expected_paths
+      LET expected_paths_lookup <= SELECT ProcessName AS ProcName, 
+        array(a1={SELECT ExpectedPath FROM expected_paths WHERE ProcessName=ProcName}) AS ExpectedPaths FROM expected_paths_tmp
+    
+      LET running_processes <= SELECT Pid AS PID, Name AS ProcessName, Ppid AS PPID, 
+        Exe AS ImagePath, CommandLine, Username, TokenIsElevated, StartTime, 
+        if(condition=EndTime<StartTime, then="", else=EndTime) AS EndTime
+        FROM process_tracker_pslist()
+        WHERE lowcase(string=Name) IN expected_paths.ProcName
+        
+
+      LET suspicious_processes <= SELECT * FROM foreach(
+          row=expected_paths_lookup,
+          query={
+              SELECT * FROM running_processes 
+              WHERE ImagePath!="" AND lowcase(string=ProcessName)=ProcName AND NOT lowcase(string=ImagePath) IN ExpectedPaths
+          }
+      )
+
+      SELECT PID, ProcessName, ImagePath, CommandLine, Username, TokenIsElevated, StartTime, EndTime, 
+        PPID, process_tracker_callchain(id=PID)[-2].Data.Name As ParentProcessName, 
+        process_tracker_callchain(id=PID)[-2].Data.Exe As ParentImagePath, 
+        process_tracker_callchain(id=PID)[-2].Data.CommandLine As ParentCommandLine, 
+        process_tracker_callchain(id=PID)[-2].Data.Username As ParentUsername,
+        process_tracker_callchain(id=PID)[-2].Data.TokenIsElevated As ParentTokenIsElevated, 
+        process_tracker_callchain(id=PID)[-2].StartTime As ParentStartTime, 
+        if(condition=process_tracker_callchain(id=PID)[-2].EndTime<StartTime, then="", else=process_tracker_callchain(id=PID)[-2].EndTime) As ParentEndTime,
```

Comment:
Why do we check the endtime < start time all the time? it is not a good idea to change the types of the data randomly - better to return a NULL or indeed the endtime.